File: Cursos/MasterLogicaProgramacion/EjerciciosPython/Ej11.py
def censura(texto, busqueda):
    textoVacio = vacio(texto)
    busquedaVacia = vacio(busqueda)
    censura = "[-CENSURADO-]"

    error = ("Texto vacío " if textoVacio else "") + ("Búsqueda vacía" if busquedaVacia else "")

    if (error != ""):
        return error

    texto = texto.lower()
    busqueda = busqueda.lower()

    textoCensuradoList = texto.split(" ")
    textoCensurado = " ".join([censura if a == busqueda else a for a in textoCensuradoList])

    return textoCensurado

def vacio(texto):
    return (texto == None or texto == "")

print(censura("HOLA QUE TAL HOLA HOLA HOLA", "HOLA"))

File: Cursos/MasterLogicaProgramacion/EjerciciosJavascript/Ej11.js
function censura(texto, busqueda) {
    let textoVacio = vacio(texto);
    let busquedaVacia = vacio(busqueda);
    let censura = "[-CENSURADO-]";

    let error = (textoVacio ? "Texto vacío " : "") + (busquedaVacia ? "Búsqueda vacía" : "");

    if (error != "") {
        return error;
    }

    texto = texto.toUpperCase();
    busqueda = busqueda.toUpperCase();

    let textoCensurado = texto.split(" ")
        .map((x) => {
         if (x == busqueda){
            return censura;
         }
         return x;
        }).join(" ");

    return textoCensurado;
}

function vacio(texto){
    return (texto == null || texto == "");
}

console.log(censura("HOLA QUE TAL HOLA HOLA HOLA", "HOLA"));